pkg/reconciler/ingress/controller.go

```diff
@@ -148,8 +148,7 @@ func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
 	// until a Knative service is deployed. This is important because the
 	// gateway pods will not be marked as healthy until they have been able to
 	// fetch a config.
-	r.caches.AddStatusVirtualHost()
-	err := r.caches.SetListeners(kubernetesClient)
+	err := r.caches.InitConfig(kubernetesClient, r.extAuthz)
```

Comment:
Should the initialization be part of `caches.new()` ?
Otherwise, it forces the caller to call InitConfig always after caches.new(), something that we might forget at some point.